crates/python/qcs_sdk/qpu/experimental.pyi

```diff
@@ -0,0 +1,165 @@
+    >>> program.add_instruction(Instruction.from_pragma(pragma_extern))
+    >>> call = Call(ChooseRandomRealSubRegions.NAME, [
+    ...     CallArgument.from_identifier("destination"),
+    ...     CallArgument.from_identifier("source"),
+    ...     CallArgument.from_immediate(complex(3, 0)),
+    ...     CallArgument.from_memory_reference(MemoryReference("seed", 0)),
+    ... ])
+    >>> program.add_instruction(Instruction.from_call(call))
+    >>> print(program.to_quil())
+    PRAGMA EXTERN choose_random_real_sub_regions "(destination : mut REAL[], source : REAL[], sub_region_size : INTEGER, seed : mut INTEGER)"
+    DECLARE destination REAL[3]
+    DECLARE source REAL[9]
+    DECLARE seed INTEGER[1]
+    CALL choose_random_real_sub_regions destination source 3 seed[0]
+
+    From there, you may reference the `destination` memory region in your pulse program.
+    """
+
+    NAME: str
+    """
+    The name of the extern call function, which may be used in `PRAGMA EXTERN` and `CALL` instructions.
+    """
+
+    @classmethod
+    def build_signature(cls) -> str:
+        """Build the Quil signature of the `PRAGMA EXTERN` instruction."""
+        ...
+
+def lfsr_v1_next(seed_value: PrngSeedValue) -> PrngSeedValue:
+    """Given a seed value, return the next value in the LFSR v1 PRNG sequence."""
+    ...
+
+def choose_random_real_sub_region_indices(
+    seed: PrngSeedValue,
+    start_index: int,
+    series_length: int,
+    sub_region_count: int,
+) -> List[int]:
+    """
+    Given a seed value, the starting index and length of a pseudo-random series, and the number of
+    sub-regions from which to choose, return a list of the sub-region indices that were chosen.
+
+    The LFSR v1 pseudo-random number generator underlies this sequence.
+    """
+    ...
+
+@final
+class RandomizedMeasurementsError(ValueError):
+    """
+    An error that can occur when adding randomized measurements to a program.
+    """
+
+    ...
+
+@final
+class UnitarySet:
+    """
+    A set of unitaries that may be applied to a pulse program prior to measurement. Currently,
+    there is a single enum variant, `ZXZXZ`, which represents a unitary as the following
+    sequence of instructions:
+
+    RZ(angle_0)-RX(pi/2)-RZ(angle_1)-RX(pi/2)-RZ(angle_2)
+
+    Each unitary is, thus, represented as the three angles which parameterize the RZ gates.
+    """
+
+    def is_zxzxz(self) -> bool: ...
+    def as_zxzxz(self) -> Optional[NDArray[np.float64]]: ...
+    def to_zxzxz(self) -> NDArray[np.float64]: ...
+    @staticmethod
+    def from_zxzxz(inner: NDArray[np.float64]) -> "UnitarySet":
+        """
+        Initialize a `UnitarySet` as a ZXZXZ decomposition. The input array must have shape (n, 3),
+        where n is the number of unitaries in the set.
+        """
+        ...
+
+@final
+class RandomizedMeasurement:
```

Comment:
This class is a `qcs-sdk` substitute for `quil-py`'s `Measurement`. Again, we can't include `quil-rs` types in the interface without providing a `quil-py` module within `qcs-sdk`.